### ISSUE TITLE
HTTPS filtering in AdGuard 4.0 (preview versions) still broken

### ISSUE THREAD

**Issue by engelhro** [AdguardTeam/AdguardForAndroid#4768]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.0 RC1

### Environment

```markdown
- OS: Android 12 (OxygenOS 12.1)
- Device: OnePlus 7 Pro
```

### Root access

- [ ] Yes, I have it.

### What filters do you have enabled?

AdGuard Base filter, AdGuard Mobile Ads filter, AdGuard German filter, AdGuard Social Media filter, AdGuard Annoyances filter

### Issue Details

After upgrading to Android 4.0 beta, I experience problems with retrieving data in any app, at least initially. A second attempt (e.g. a reload) helps in most cases to establish the connection.

I guess it's the same problem described in issue #4639, but that was closed as "solved" (resp. the problem has not occurred again). Well, for me it still does…

Some more details:

1. The problem is not restricted to browsers alone, **many types of apps** are affected (e.g. also F-Droid when downloading the repository information, any news client retrieving data from sources like Reddit/HackerNews, etc.).
2. Apps react very differently to the failed inital connection attempt. Many just allow a reload or second attempt (which then succeeds), but a few insist on their "no connection" error message and can't be convinced to try again.
3. My assumption that the issue is the same as (or closely related to) issue #4639 is based on the fact that disabling HTTPS filtering indeed solves the problem. An ugly workaround, but at least one that works. I would've never found that out myself, I always completely disabled AdGuard so far for the stubborn apps.

The problem appeared after migration from AdGuard 3.x final to the 4.0 beta series. The problem existed in the first beta, then in a second beta, and now continues to appear also in RC1 without being addressed properly so far (the statement "We have made some adjustments" in issue #4639 is *extremely vague* and doesn't sound like an actual bug being found and fixed).

I'd really like to see that issue solved prior to the final AdGuard 4.0 release, as using AdGuard without HTTPS filtering is annoying.

I can still later try to completely reset AdGuard and reinitialize everything so I have a fresh configuration (but that would affect the statistics, and I'd be happy if I could keep them 😉)

### Expected Behavior

Behaviour as with the current AdGuard 3.x series: a working implementation not impacting the connection/download attempts of apps using HTTPs.

### Actual Behavior

Details are given in "Issue Details" above.

As this doesn't seem to be a widespread issue and only affects a few users, maybe it's related to the upgrade to 4.0 beta from the stable 3.x versions? Something odd when converting the settings? Just a speculation though.

### Screenshots

<details><summary>Screenshot 1:</summary>

</details>

**Comment by engelhro** [comment]
Update, just for information: the problem persists, even with the now released final version 4.0 of Adguard for Android…

**Comment by Versty** [comment]
@engelhro Hi! 

Take my apologies for delayed response. This issue dropped out of our sight for a while. 
Let's see if we could fix the problem. So if the issue still persists with the latest release version of AdGuard for Android, please collect the debug logs from your device by following the steps below.

1.  Enable debug logging: 
*Settings -> General -> Advanced -> Logging level -> Debug*

2. Reproduce the issue, then remember the exact time when it happened 

3. Collect logs: 
*Settings -> General -> Advanced -> Export logs and system info* 

4. Send this file to devteam@adguard.com:
      - mention the `4768` number in the subject
      - specify the exact time when the issue occurred

**Comment by Versty** [comment]
@engelhro Does the issue still persist?

**Comment by engelhro** [comment]
Hello,

The underlying problem probably still exists, but waiting with degraded Adguard functionality for a fix in an issue not being answered for two months was not an option…😕

So, as there's no possibility to fully backup Adguard settings (i.e., including the stats!), I decided _to fully reset the app anyway and setup everything again manually_ (without using a backup). As far as I'm aware I reproduced all or at least most settings in all screens as they have been before.

**The issue is gone now, the fresh Adguard has no problems with apps not immediately connecting**. So at least I cannot help to track down this bug and to reproduce it – sorry!

I guess this issue can be closed then, solved by timeout.
